Lib/unittest/test/testmock/testasync.py

```diff
@@ -811,12 +828,10 @@ def test_assert_awaited_but_not_called(self):
             self.mock.assert_awaited()
         with self.assertRaises(AssertionError):
             self.mock.assert_called()
-        with self.assertRaises(TypeError):
-            # You cannot await an AsyncMock, it must be a coroutine
```

Comment:
As per this comment I guess not implementing `__await__` was a conscious decision. This PR adds a behavior change making `AsyncMock` awaitable. I will leave this to @lisroach since I forgot the reason why this was decision was made. Perhaps I would suggest opening a new issue to track this and revert changes that make AsyncMock awaitable. For reference this comment was added in ef048517755db1f0d211fb6dfc655a8b412cc96f and discussion at bpo-38136 .